### PR TITLE
nova reclaim, https novnc, and autoevacuate

### PR DESCRIPTION
**Pull request by Farice4**
```
  delete reclaim instance only if configure
  update reclaim instance interval time 10 days
```

Signed-off-by: apporc appleorchard2000@gmail.com

Conflicts:
    puppet/modules/eayunstack/manifests/upgrade/nova.pp

### CODE REVIEW

File: puppet/modules/eayunstack/manifests/upgrade/nova.pp [hunk 1]
```diff
@@ -6,7 +6,7 @@
   $admin_password = $fuel_settings['cinder']['user_password']
   $admin_tenant_name = 'services'
   # reclaim interval is 15 days.
```
Comment (huntxu) [inline puppet/modules/eayunstack/manifests/upgrade/nova.pp]: 注释这行没改

**Review by huntxu — CHANGES_REQUESTED** [review]

File: puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      }
+
+      file { 'consul_config_file':
+        ensure  => file,
+        path    => "/etc/consul/storage/consul.json",
+        require => File['consul_config_dictory'],
+        content => template('eayunstack/consul.erb')
+      }
+
+      file { 'consul_exec_dictory':
```
Comment (huntxu) [inline puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp]: 这是一个文件，但是命名却是"dictory"？还是是一个目录但是内容写错了？

**Comment by huntxu** [comment]
第三个commit有以下问题，基本都和引号的使用有关，
注意一下第10行那里完全不需要使用引号和花括号包围单一一个变量
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 28
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 34
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 41
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 48
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - ERROR: single quoted string containing a variable found on line 58

**Review by huntxu — CHANGES_REQUESTED** [review]
第三个commit有以下问题，基本都和引号的使用有关，
注意一下第10行那里完全不需要使用引号和花括号包围单一一个变量
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 28
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 34
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 41
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: double quoted string containing no variables on line 48
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - WARNING: string containing only a variable on line 10
puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - ERROR: single quoted string containing a variable found on line 58

File: puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp (hunk 2 above)
Comment (Farice4) [reply puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp]: 是一个文件，命名错误，已修改

File: puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        require => Package['consul'],
+        owner   => 'consul',
+        group   => 'consul',
+      }
+
+    augeas { 'evacuate_config_file':
+        context => '/files/etc/autoevacuate/evacuate.conf',
+        lens    => 'Puppet.lns',
+        incl    => '/etc/autoevacuate/evacuate.conf',
+        changes => 'set novaclient/auth_url ${auth_url}',
```
Comment (huntxu) [inline puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp]: 这里仍有错误，字符串里有变量应该有双引号
`puppet/modules/eayunstack/manifests/upgrade/auto_evacuate/auto_evacuate.pp - ERROR: single quoted string containing a variable found on line 56`

**Review by huntxu — CHANGES_REQUESTED** [review]

**Pull request merged**